### ISSUE TITLE
OpenTelemetry.AutoInstrumentation fails with console application

### ISSUE THREAD

**Issue by wondering639** [open-telemetry/opentelemetry-dotnet-instrumentation#3911]
# Bug Report

## Symptom

**Describe the bug**
Install the NuGet package into a "regular" (no ASP.NET or so) console application, following instructions at https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/using-the-nuget-packages.md

**Expected behavior**
Working instrumentation, however I am unable to start my instrumented application:
```
You must install or update .NET to run this application.

App: /app/MyApp.dll
Architecture: x64
Framework: 'Microsoft.AspNetCore.App', version '9.0.0' (x64)  <================ Note ASP.NET Core here, though I have a "regular" console application
.NET location: /usr/share/dotnet/

No frameworks were found.
```
Note the framework Microsoft.**AspNetCore**.App while I have a "regular" console application. Instrumentation seems to change this. Therefore it is unable to start the application, as only the regular runtime is installed in `mcr.microsoft.com/dotnet/runtime:9.0`.
This is what is installed in the container and is sufficient when running without instrumentation:
```
app@image:/app$ dotnet --list-runtimes
Microsoft.NETCore.App 9.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

Side note: Interestingly, when using the "shell script installer" https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation?tab=readme-ov-file#shell-scripts everything works. However, as installing the NuGet package is the recommended way and I have access to the source code, I prefer to install the NuGet package instead of using the installer.

**Screenshots**

**Runtime environment (please complete the following information):**

- OpenTelemetry Automatic Instrumentation version: OpenTelemetry.AutoInstrumentation 1.9.0
- OS: Docker image mcr.microsoft.com/dotnet/runtime:9.0
- .NET version: 9

**Additional context**

## Reproduce

 1. Create console application, including Host.CreateApplicationBuilder; based on `worker` template
 2. Install OpenTelemetry.AutoInstrumentation
 3. Run, in my case inside docker.
 4. Note it will fail to run with instrumentation, expecting a ASP.Net runtime. See *Expected behavior* for details.

**Comment by wondering639** [comment]
Did some further testing with a new project without docker. This makes it also easier to demonstrate the issue:
1. dotnet new worker
The dependencies in Jetbrains Rider now look like this:
![image](https://github.com/user-attachments/assets/156c1dd1-a0cf-4016-ae3b-0060a227cfc9)

Once I add `OpenTelemetry.AutoInstrumentation`, they look like this:
![image](https://github.com/user-attachments/assets/461e84f3-b26a-4f3b-8dcf-8d15c668b9ee)
Note how frameworks now additionally contains `Microsoft.AspNetCore.App`.

**Comment by RassK** [comment]
May try to skip the ASP.NET Core instrumentation
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/using-the-nuget-packages.md#using-the-nuget-packages

... although for sure it's not going to change installed deps.

**Comment by wondering639** [comment]
yeah, I think the instrumentation is not the issue, but the dependency on `Microsoft.AspNetCore.App`. Interestingly the script version seems to have no issues with that, so I hope this can be optimized.

**Comment by RassK** [comment]
Yes, just the AspNetCore instrumentation should be lazy loaded. At least on the runtime it should not crash then.
NuGet package has to ship with certain dependency requirements, that should be the difference.

**Comment by RassK** [comment]
Unfortunately these are not the only issues. It will bring anyway max set of packages and the dependencies (some by the reference and not physically).

**Comment by pjanotti** [comment]
Confirmed that adding `PrivateAssets="all"` to the `FrameworkReference` than the application doesn't get `Microsoft.AspNetCore.App` added to the `*.runtimeconfig.json` file. If this was to be the fix we would have to add the attribute to the two packages listed above.

**Comment by wondering639** [comment]
> I hate bumping but this is a showstopper for using this library for me, I don't want to make hosts pull another framework for instrumentation :/

you can use the shell script installer as workaround https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation?tab=readme-ov-file#shell-scripts
Your bumping is justified nevertheless ;-)

**Comment by pjanotti** [comment]
Per SIG meeting: let's try to see if it is possible to repackage the assemblies from `OpenTelemetry.Instrumentation.AspNetCore` while setting `PrivateAssets="all"` - there are other ideas, but, this seems the best compromise at this point.

cc @aragia

**Comment by iskiselev** [comment]
Should `OpenTelemetry.Instrumentation.AspNetCore` (or probably even `OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper`) be optional and included separately in a similar way to `OpenTelemetry.Instrumentation.StackExchangeRedis` with validation though [MsBuild CheckForInstrumentationPackagesTarget](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/4faf86d1b2c78c2ee9a0cd87edb134fd4512b841/src/OpenTelemetry.AutoInstrumentation.BuildTasks/OpenTelemetry.AutoInstrumentation.BuildTasks.targets#L5)?

**Comment by iskiselev** [comment]
Discussed #5029  at SIG, desicion was to not proceed at this time. Instead next workaround can be applied on projects that reffers us:
```xml
  <Target Name="RemoveAspNetCoreFrameworkReference" AfterTargets="ResolvePackageAssets">
    <ItemGroup>
      <TransitiveFrameworkReference Remove="Microsoft.AspNetCore.App" />
    </ItemGroup>
  </Target>
```
